Configuration/UTMConfigurable.h

```diff
@@ -28,6 +35,9 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, nullable, copy) NSString *notes;
 
 @property (nonatomic, nullable, copy) NSString *consoleTheme;
+// TODO: Maybe use CGColor (But hard to handle)
+@property (nonatomic, nullable, copy) PlatformColor *consoleTextColor;
+@property (nonatomic, nullable, copy) PlatformColor *consoleBackgroundColor;
```

Comment:
I was thinking maybe using a `CGColor` might be a good idea for two reasons:
- SwiftUI's `ColorPicker` can take a binding to a `CGColor`
- It would be compatible with both platforms, the `PlatformColor` stuff wouldn't be needed

But the problem is that `CGColor` isn't an object, so passing it around is harder. And also I'm not sure if `CGColor` can be used with a keyed archiver to save it.

I think it wouldn't be worth the extra trouble, since it isn't that much code in the `Binding` that converts a `NSColor` to the SwiftUI `Color` type.

Reply:
> But the problem is that CGColor isn't an object, so passing it around is harder.

Why is it any harder than say NSInteger?

Reply:
@osy How would it be done? (What should the property attributes be and do I use `struct CGColor *` or `CGColorRef *`)?

Reply:
Look at my latest comment, I think NSNumber or NSString would be better fit here.

Reply:
@osy The alternative solution would be to store an NSColor somewhere in UTMConfiguration, and then convert it to a hex string when saving the plist, like how it's done for the Apple backend.

If you want to do that, how/where should the intermediate NSColor be stored, and where can I put the conversion code?

Reply:
I think it should be NSString in UTMConfigurable and both QEMU and Apple backends store the hex string.

Then when it's used for QEMU terminal, it uses the hex string directly. For Swift terminal it converts the hex string to NSColor (through your extension) in the view controller when it's being set.

Reply:
@osy How do we prevent the drift then?

Reply:
Storage call .hexString

Usage call NSColor(fromHexString: String) which you implemented in the extension but you need to change the init there to call init(srgbRed:green:blue:alpha)

Reply:
@osy You see, I can do that for the Apple backend because it's only converted when storing, do drift isn't a problem when selecting the color.

It's different with the Qemu backend because when setting or getting the console colors, it will have to do the conversion to save it into the dictionary. And because it's directly bound to the value used in the SwiftUI color picker, a small amount of drift on each conversion will be amplified to the point of not being usable.

Reply:
Can you try this different multiplier https://stackoverflow.com/a/39358741 seems like UIColor uses extended sRGB

Reply:
@osy Nope, there's still a (slightly smaller but still noticeable/unusable) amount of drift.

I think the only solution if you want to store the console colors as a hex string is to only do the conversion when saving/loading the plist, not in the setter/getter for the color properties.

Reply:
Any chance you can ask stackoverflow with details on what you're seeing? I don't think there should be drift and maybe it's a bug that others have workarounds for.

Reply:
@osy Done: https://stackoverflow.com/questions/71016043/swiftui-colorpicker-with-binding-to-hex-string-causes-large-drift